### ISSUE TITLE
ChatGPT infinite checking browser

### ISSUE THREAD

**Issue by ollidoetsch** [ferdium/ferdium-app#1043]
### Avoid duplicates

- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22bug+%3Abeetle%3A%22) for a bug report similar to mine, in vain

### Ferdium Version

6.2.4

### What Operating System are you using?

Windows

### Operating System Version

Windows 10 Home version 1903

### What arch are you using?

x64

### Last Known Working Ferdium version

Didn't used ChatGPT before in Ferdium

### Expected Behavior

ChatGPT should just open like on a regular browser

### Actual Behavior

When opening ChatGPT in Ferdium it needs to verify that you're a human by letting you click a checkbox, then it loads to check the browser but always shows the checkbox again. It's an infinite loop.

### Steps to reproduce

1. Add ChatGPT as Service to Ferdium
2. Open in and click the "Verify that you're a human" checkbox
3. See if the checkbox needs to be clicked infinite

### Debug link

https://debug.ferdium.org/0b341f09-c223-4058-8e7d-c0fec77fe97a

### Screenshots

![Step 1](https://user-images.githubusercontent.com/10545005/223675706-8284eadd-8daa-45b6-8bc1-4bf260dac165.jpg)
![Step 2](https://user-images.githubusercontent.com/10545005/223675710-6b3ca45e-ecf2-4428-849a-72bb2250ee9e.jpg)
![Step 3](https://user-images.githubusercontent.com/10545005/223675714-4f404731-2fa4-44f1-83fb-ba47dd3e5d54.jpg)

**Comment by yojo3000** [comment]
modify the url in package.json to this instead:

https://chat.openai.com/auth/login

and the browser checking will end after a few tries
at least it works for me

**Comment by Gaibhne** [comment]
> modify the url in package.json to this instead:
> 
> https://chat.openai.com/auth/login
> 
> and the browser checking will end after a few tries at least it works for me

That has not helped for me; after editing the file I tried to restart Ferdium, reload Ferdium, reload services, remove the service and readd it in several orders, Nothing helped, I'm still stuck with the endless loop. The file I changed was `C:\Users\me\AppData\Roaming\Ferdium\recipes\chatgpt\package.json` - is that the correct one ?

Edit: adding a custom service with `https://chat.openai.com/auth/login` works fine, so that can be used as a workaround until the proper service is fixed.

**Comment by ollidoetsch** [comment]
Hey guys, unfortunately both solutions didn't work for me. When creating the custom service I just copied the whole ChatGPT folder and changed the URL in the .json file. Is that correct or do I need to add it somehow else?

**Comment by JMVS** [comment]
None of the solutions worked for me either...any other workarounds?

EDIT: May is TLS fingerprinting related? 
I found this: https://www.zenrows.com/blog/what-is-tls-fingerprint#how-to-bypass-tls-fingerprinting
But I don't know if/how could be implemented.

**Comment by lampros-m** [comment]
Same problem here. Both workarounds don't work for me.

**Comment by Zaczero** [comment]
> None of the solutions worked for me either...any other workarounds?
> 
> EDIT: May is TLS fingerprinting related? I found this: https://www.zenrows.com/blog/what-is-tls-fingerprint#how-to-bypass-tls-fingerprinting But I don't know if/how could be implemented.

Performing any sort of bypass is not a proper long-term solution. I believe this bug should be reported to Cloudflare as they are preventing Ferdium from accessing ChatGPT.

In my case, it worked to simply pass the challenge about 10 times repeatedly. I just spammed it until I got lucky. :-)

**Comment by Lars-** [comment]
Same problem here...

**Comment by martin-braun** [comment]
@Zaczero How is this the responsibility of CloudFlare to make Ferdium compatible? Any mainstream browser will be accepted, so Ferdium fails to appear like a browser in that sense.

**Comment by Zaczero** [comment]
> How is this the responsibility of CloudFlare to make Ferdium compatible

Because this is a false-positive, they are blocking something that shouldn't be blocked. They are fingerprinting the client and noticing that it's unusual, so they block it. It's just a matter of them adding an appropriate rule for Ferdium users.

**Comment by JMVS** [comment]
> > How is this the responsibility of CloudFlare to make Ferdium compatible
> 
> Because this is a false-positive, they are blocking something that shouldn't be blocked. They are fingerprinting the client and noticing that it's unusual, so they block it. It's just a matter of them adding an appropriate rule for Ferdium users.

Yes, I think the same. Is there any way for us users to let CloudFlare know about this?

**Comment by Zaczero** [comment]
I did go through their contact page but in the end they told me there is no direct support for free users.

![image](https://user-images.githubusercontent.com/10835147/228073605-dc68d80e-e9b7-4f23-86e5-94d64e31320b.png)

I suppose we should ask on the community forums, so I did:
https://community.cloudflare.com/t/ferdium-stuck-in-an-infinite-captcha-loop/489705


I did search for false-positives and found this page: https://developers.cloudflare.com/ddos-protection/managed-rulesets/adjust-rules/false-positive/ - It explains how to handle a false-positive for a website owner (whitelist rule) so it's not really applicable here.

Even if OpenAI were to add this whitelist rule, there's potentially more sites that would need fixing so I don't see this as a complete fix. It would be best to resolve this issue at its core:

The core issue is that even if we try to complete the captcha challenge, we are unable to do so. Stuck in an infinite loop.

**Comment by JMVS** [comment]
For what is worth, using this Ferdium version:
Version: 6.2.6
Server: No Server
Electron: 23.2.0
Chrome: 110.0.5481.192
Node.js: 18.12.1

I was able to log into ChatGPT after 6 tries at the captcha.

**Comment by Lars-** [comment]
It indeed works after 10 tries here. But then the next (possibly related) problem arises, I can't login with Google because "This browser or app may not be secure. Try using a different browser. If you’re already using a supported browser, you can refresh your screen and try again to sign in."...

**Comment by Lars-** [comment]
Update: I got it to work using the browser service. Google showed another login prompt using that service. It resulted in an OAuth error a couple of times, but then it suddenly worked. I have ChatGPT finally working in Ferdium 🎉

**Comment by adromir** [comment]
can you describe how you achieved that 

> Update: I got it to work using the browser service. Google showed another login prompt using that service. It resulted in an OAuth error a couple of times, but then it suddenly worked. I have ChatGPT finally working in Ferdium 🎉

**Comment by SpecialAro** [comment]
Hey!

Have you attempted to clean the cache of the service (`Edit > Clear cache`) ?
You can also manually open the `View > Toggle Service Developer Tools`, go to the `Application` tab and open (each element of the list) and right click on each element (inside the element of the list):
- Local Storage - `Right Click > Clear`
- Session Storage - `Right Click > Clear`
- Cookies - `Right Click > Clear`

After this, reload Ferdium/the recipe and it should negociate new tokens with Cloudflare.

About the Google login, I've made a PR that hopefully gets merged tomorrow so it releases in tomorrow's nightly that fixes that issue (https://github.com/ferdium/ferdium-recipes/pull/413)

Let me know if this fixes it for you and if we can close this issue. Thank you!
